### PR TITLE
[WIP] Boolean support

### PR DESCRIPTION
**Pull request by zabulus**
1. Added Tests ported from Jaybird.
2. Added support for reading values from database. GDS

Stuck on writing parameters to wire.

### CODE REVIEW

File: NETProvider/src/FirebirdSql.Data.FirebirdClient/Client/Managed/Version10/GdsStatement.cs [hunk 1]
```diff
@@ -413,8 +413,6 @@ public override DbValue[] Fetch()
 			}
 			else
 			{
-				_rows.Clear();
```
Comment (zabulus) [inline NETProvider/src/FirebirdSql.Data.FirebirdClient/Client/Managed/Version10/GdsStatement.cs]: _rows already empty at this point

File: NETProvider/src/FirebirdSql.Data.FirebirdClient/Client/Managed/Version10/XdrStream.cs [hunk 2]
```diff
@@ -466,6 +466,13 @@ public object ReadValue(DbField field)
 				case DbDataType.TimeStamp:
 					fieldValue = ReadDateTime();
 					break;
+
+				case DbDataType.Boolean:
+					fieldValue = ReadInt32() != 0;
```
Comment (zabulus) [inline NETProvider/src/FirebirdSql.Data.FirebirdClient/Client/Managed/Version10/XdrStream.cs]: Don't know why it takes 4 bytes on the wire.

File: NETProvider/src/FirebirdSql.Data.FirebirdClient/Common/DbValue.cs [hunk 3]
```diff
@@ -101,6 +101,11 @@ public char GetChar()
 
 		public bool GetBoolean()
 		{
+			if(this.IsDBNull())
```
Comment (zabulus) [inline NETProvider/src/FirebirdSql.Data.FirebirdClient/Common/DbValue.cs]: Also not sure about this.

**Comment by cincuranet** [comment]
Some pieces merged in branch [`bool`](../tree/bool) and [`bool-tests`](../tree/bool-tests). Will eventually and up in `master`.

**Pull request closed without merging**